### ISSUE TITLE
[Flutter web-semantics] Feature request: Enable Accessibility button upper left 1 pixel instead of offscreen 1 pixel

### ISSUE THREAD

**Issue by ecolab-neil** [flutter/flutter#177670]
### Use case

Automation tools such as Maestro and probably Appium seem to not be able to tap on
`      document.querySelector('flt-semantics-placeholder[aria-label="Enable accessibility"]')
                .click();` because it is at absolute positivion -1 left and -1 top of the screen (offscreen).

Maestro does appear to be able to tap on an invisible button with no text if it is at 1,1.

I am not sure if i have bandwidth to PR this but i wanted to get the discussion going.
See: https://docs.flutter.dev/ui/accessibility/web-accessibility#invisible-button

Can anyone think of a reason that the ('flt-semantics-placeholder[aria-label="Enable accessibility"]' has to be offscreen rather than "just" onepixel, invisible, transparent etc?

### Proposal

Change just 2 characters in the code to make the button be "+" 1 left and "+" one top insteda of "-" (or remove the negative)

**Comment by flutter-zl** [comment]
@ecolab-neil , Thank you for the detailed feedback!

### Why Not Move Button to (1,1)?
I actually implemented and tested moving the button from (-1,-1) to (1,1) to determine the impact. Here's what I found:

**Critical Issue: Tab Order Disruption**
The button has tabindex="0" by design. From [the code comments](https://github.com/flutter/flutter/blob/master/engine/src/flutter/lib/web_ui/lib/src/engine/semantics/semantics_helper.dart#L204C1-L205C1) :
 `'tabindex' makes sure the button is the first target of tab`

**At (1, 1) onscreen:**
Still "first tab target" but now at a visible position.
When users press Tab, focus moves to top-left 1px invisible element.
Accessibility audits flag: "Focusable element with no visible focus indicator".
Violates WCAG 2.4.7 (Focus Visible) - focus indicator must be visible.

**Could We Remove tabindex?**
Removing tabindex="0" would solve the tab order issue, but then keyboard-only users couldn't discover the button at all - breaking the original design intent.

### Recommended Solution: Query Parameter
Instead, I propose an engine-level query parameter:

In platform_dispatcher.dart, check for a query parameter during initialization:
`if (Uri.base.queryParameters.containsKey('flutter-semantics')) {
  scheduleMicrotask(() {
    EngineSemantics.instance.semanticsEnabled = true;
  });
}
`

**Usage**
Automation tools simply append the query parameter to the URL:
https://example.com/?flutter-semantics
Works with hash routing too:
https://example.com/?flutter-semantics#/login

### Implementation Status
I have a working implementation in PR(https://github.com/flutter/flutter/pull/178224) and have tested it successfully. 

Please let me know your thoughts. @mdebbar  @ecolab-neil

**Comment by ecolab-neil** [comment]
@flutter-zl  this is extremely promising and i'm very optimistic/excited about it, i have two quick thoughts
1) It may need some tests and documentation
2) Does it "work fine" with navigation - we tried something similar and had troubles, although we didn't try it at the engine level; does the semantics tree "go away" every time or almost everytime you navigate to another flutter route because the url changes?  {Edit: I see your comment about hash routing which is .. similar/related to what i was asking)
I certainly wouldn't want to block this pr though, this is awesome :) 
thank you
